pkg/products/monitoringspec/reconciler.go

```diff
@@ -0,0 +1,407 @@
+			if err != nil && k8serr.IsNotFound(err) {
+				logrus.Infof("Spec phase completed")
+				//namespace is gone, return complete
+				return integreatlyv1alpha1.PhaseCompleted, nil
+			}
+			phase, err := resources.RemoveNamespace(ctx, installation, serverClient, r.Config.GetNamespace())
+			if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+				logrus.Infof("Spec phase removal failure: %v", err)
+				return phase, err
+			}
+			return integreatlyv1alpha1.PhaseInProgress, nil
+		},
+	)
+
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		logrus.Errorf("failed to reconcile finalizer: %v", err)
+		events.HandleError(r.recorder, installation, phase, "Failed to reconcile finalizer", err)
+		return phase, err
+	}
+
+	phase, err = r.createNamespace(ctx, serverClient, installation)
+	logrus.Infof("Phase: %s createNamespace", phase)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		logrus.Errorf("failed to create namespace : %v", err)
+		events.HandleError(r.recorder, installation, phase, "Failed to create namespace", err)
+		return phase, err
+	}
+
+	phase, err = r.reconcileMonitoring(ctx, serverClient, installation)
+	logrus.Infof("Phase: %s reconcileMonitoring", phase)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		logrus.Errorf("ailed to reconcile: %v", err)
+		events.HandleError(r.recorder, installation, phase, "Failed to reconcile:", err)
+		return phase, err
+	}
+
+	product.Host = r.Config.GetHost()
+	product.Version = r.Config.GetProductVersion()
+
+	err = r.ConfigManager.WriteConfig(r.Config)
+	if err != nil {
+		logrus.Errorf("failed to write config: %v ", err)
+		events.HandleError(r.recorder, installation, integreatlyv1alpha1.PhaseFailed, "Failed to update monitoring config", err)
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("could not update monitoring config: %w", err)
+	}
+
+	events.HandleProductComplete(r.recorder, installation, integreatlyv1alpha1.MonitoringStage, r.Config.GetProductName())
+	logrus.Infof("%s installation is reconciled successfully", packageName)
+	return integreatlyv1alpha1.PhaseCompleted, nil
+}
+
+// make the federation namespace discoverable by cluster monitoring
+func (r *Reconciler) createNamespace(ctx context.Context, serverClient k8sclient.Client,
+	installation *integreatlyv1alpha1.RHMI) (integreatlyv1alpha1.StatusPhase, error) {
+	namespace, err := resources.GetNS(ctx, r.Config.GetNamespace(), serverClient)
+	if err != nil {
+		if !k8serr.IsNotFound(err) {
+			return integreatlyv1alpha1.PhaseFailed, err
+		}
+		//TODO - addClusterMonitoringLabel flipped back to true
+		_, err := resources.CreateNSWithProjectRequest(ctx, r.Config.GetNamespace(),
+			serverClient, installation, false, false)
+		if err != nil {
+			return integreatlyv1alpha1.PhaseFailed, err
+		}
+		return integreatlyv1alpha1.PhaseCompleted, nil
+	}
+
+	resources.PrepareObject(namespace, installation, false, true)
```

Comment:
```suggestion
	resources.PrepareObject(namespace, installation, false, false)
```

This got changed in the most recent commit somehow

Reply:
I had flipped to true during testing. flipped back.

Reply:
VS Code needs to be smart enough , it liked having "v1" instead of "monitoringv1". fixed it now.

Reply:
> VS Code needs to be smart enough , it liked having "v1" instead of "monitoringv1". fixed it now.

What's worse is that when there are two or more imports with the package name "v1", then the second one will get the automatic alias "v12", the third "v13" and so on. I'm not sure if it's VS Code, or Goland or gopls or something else that does that.